### PR TITLE
Adds troubleshooting to GMP Integration guide

### PR DESCRIPTION
**Pull request by bstuder99**
## Proposed Changes

Based on Support ticket where clusters would display 0% efficiency

### CODE REVIEW

File: gcp-gmp-integration.md [hunk 1]
```diff
@@ -68,4 +68,65 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+```bash
+kubectl exec -i -t -n $KUBECOST_NAMESPACE \
+  deployments/$KUBECOST_DEPLOYMENT -c cost-analyzer-frontend \
+  -- curl "0:9090/model/prometheusQuery?query=node_total_hourly_cost\{CLUSTER_ID=\"$CLUSTER_ID\"\}" \
+ |jq
+```
+
+If the above queries fail, check the following:
+
+1. Check logs of the `sigv4proxy` container (may be the Kubecost deployment or Prometheus Server deployment depending on your setup):
```
Comment (jessegoodier) [inline gcp-gmp-integration.md:105]: @kc-adawson Do you have this running like described above?
GMP doesn't call it a sigV4 proxy? Looks like it is `gke.gcr.io/prometheus-engine/frontend:v0.4.1-gke.0`
Is there any other advice you have here for troubleshooting?

Comment (kc-adawson) [reply gcp-gmp-integration.md:131]: Just catching up on this - In my sandbox, I'm setting GMP up in the way documented here. https://docs.kubecost.com/install-and-configure/install/custom-prom/gcp-gmp-integration .

In my setup, the "id" metric that would be retrieved via these debug instructions returns blank, so if people run into an "id" that is blank using the debug steps,  they can use the following to force set "cluster" as the Prom Cluster ID label

kubecostModel:
  promClusterIDLabel: cluster
  
  ```
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "",
          "cluster": "",
          "id": "/",         <<<<--- This is blank for GMP/(Autopilot?) Prometheus, need to use "cluster"
          "instance": "",
          "job": "kubelet",
          "location": "",
          "node": "",
          "project_id": ""
        },
        "value": [
          1697627020,
          "2358820864"
        ]
      },
```

**Comment by bstuder99** [comment]
Conflicts should be resolved

**Comment by bstuder99** [comment]
Going to update the doc with Andrew's suggestions first

File: gcp-gmp-integration.md (hunk 1 above)
Comment (bstuder99) [reply gcp-gmp-integration.md]: Thanks, I've added it into the current doc.

**Comment by bstuder99** [comment]
@chipzoller @jessegoodier We're ready for review if you want to look at this

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 2]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+
+```bash
+KUBECOST_NAMESPACE=kubecost
+KUBECOST_DEPLOYMENT=kubecost-cost-analyzer
+CLUSTER_ID=YOUR_CLUSTER_NAME
+```
+
+1. Verify connection to GMP and that the metric for `container_memory_working_set_bytes` is available:
+
+If you have set `kubecostModel.promClusterIDLabel`, you will need to change the query (`CLUSTER_ID`) to match the label.
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:85]:
```suggestion
If you have set `kubecostModel.promClusterIDLabel` in the Helm chart, you will need to change the query (`CLUSTER_ID`) to match the label.
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 3]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+
+```bash
+kubectl exec -i -t -n $KUBECOST_NAMESPACE \
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:88]:
```suggestion
kubectl exec -it -n $KUBECOST_NAMESPACE \
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 4]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+  deployments/$KUBECOST_DEPLOYMENT -c cost-analyzer-frontend \
+  -- curl "0:9090/model/prometheusQuery?query=container_memory_working_set_bytes\{CLUSTER_ID=\"$CLUSTER_ID\"\}" \
+ |jq
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:91]:
```suggestion
 | jq
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 5]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+```
+
+2. Verify Kubecost metrics are available in GMP:
+
+```bash
+kubectl exec -i -t -n $KUBECOST_NAMESPACE \
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:97]:
```suggestion
kubectl exec -it -n $KUBECOST_NAMESPACE \
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 6]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+  deployments/$KUBECOST_DEPLOYMENT -c cost-analyzer-frontend \
+  -- curl "0:9090/model/prometheusQuery?query=node_total_hourly_cost\{CLUSTER_ID=\"$CLUSTER_ID\"\}" \
+ |jq
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:100]:
```suggestion
 | jq
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 7]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+```
+
+You should receive an output similar to:
+
+```
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:105]:
```suggestion
```json
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 8]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+        },
+        "value": [
+          1697627020,
+          "2358820864"
+        ]
+      },
+```
+
+{% hint style="info" %}
+If `id` returns as a blank value, you can set the following Helm flah to force set `cluster` as the Prometheus cluster ID label:
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:130]:
```suggestion
If `id` returns as a blank value, you can set the following Helm value to force set `cluster` as the Prometheus cluster ID label:
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 9]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+
+```
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:132]:
```suggestion
```yaml
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 10]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+kubecostModel:
+promClusterIDLabel: cluster
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:134]:
```suggestion
  promClusterIDLabel: cluster
```

No indentation was here.

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 11]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+
+```log
+time="2023-09-21T17:40:15Z" level=info msg="Stripping headers []" StripHeaders="[]"
+time="2023-09-21T17:40:15Z" level=info msg="Listening on :8005" port=":8005"
+```
+
+2. Check logs in the `cost-model` container for Prometheus connection issues:
+
+```bash
+kubectl logs deployments/$KUBECOST_DEPLOYMENT -c cost-model --tail -1 |grep -i err
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:158]:
```suggestion
kubectl logs deployments/$KUBECOST_DEPLOYMENT -c cost-model --tail -1 | grep -i err
```

File: install-and-configure/install/custom-prom/gcp-gmp-integration.md [hunk 12]
```diff
@@ -68,4 +68,100 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
...
+```
+
+Example errors:
+
+```log
+ERR CostModel.ComputeAllocation: pod query 1 try 2 failed: avg(kube_pod_container_status_running...
+Prometheus communication error: 502 (Bad Gateway) ...
```
Comment (chipzoller) [inline install-and-configure/install/custom-prom/gcp-gmp-integration.md:165]: No ending code fence here.

Comment (bstuder99) [reply install-and-configure/install/custom-prom/gcp-gmp-integration.md:165]: I've added the code fence

**Review by jessegoodier — APPROVED** [review]

**Pull request merged**